<?xml version="1.0"?>
<launch>
    <!--<arg name="world_name" default="worlds/robocup14_spl_field.world"/>-->
    <arg name="laser_enabled" default="true"/>
    <arg name="realsense_enabled" default="false"/>
    <include file="$(find cpr_office_gazebo)/launch/office_world.launch">
        <arg name="laser_enabled" value="$(arg laser_enabled)"/>
        <arg name="realsense_enabled" value="$(arg realsense_enabled)"/>
    </include>

    <node name="husky_drive" pkg="husky_highlevel_controller_drive" type="husky_highlevel_controller_drive" output="screen">
        <rosparam command="load" file="$(find husky_highlevel_controller)/config/config.yaml" />
    </node>
    <node name="husky_ctrl" pkg="husky_highlevel_controller" type="husky_highlevel_controller" output="screen">
        <rosparam command="load" file="$(find husky_highlevel_controller)/config/config.yaml" />
    </node>
    

    <node name="teleop" pkg="teleop_twist_keyboard" type="teleop_twist_keyboard.py" output="screen">
        <remap from="cmd_vel" to="/husky_velocity_controller/cmd_vel"/>
    </node>
    <!--  Run rviz -->
    <node name="rviz" pkg="rviz" type="rviz" output="screen" args="-d $(find husky_highlevel_controller)/config/aufgabe2.rviz"/>

    <arg name="scan_topic" default="/scan"/>

    <!--- Run gmapping 
    <include file="$(find husky_navigation)/launch/gmapping.launch">
        <arg name="scan_topic" value="$(arg scan_topic)" />
    </include>-->

    <!-- Run Mapp server-->
    <arg name="map_file" default="$(find husky_highlevel_controller)/config/map.yaml"/>

    <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)" />

    <include file="$(find husky_navigation)/launch/amcl.launch">
        <arg name="scan_topic" value="$(arg scan_topic)" />
    </include>


    <include file="$(find husky_navigation)/launch/move_base.launch">
    </include>
   
</launch>
